sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala

```diff
@@ -157,7 +158,8 @@ object OrcUtils extends Logging {
         // In these cases we map the physical schema to the data schema by index.
         assert(orcFieldNames.length <= dataSchema.length, "The given data schema " +
           s"${dataSchema.catalogString} has less fields than the actual ORC physical schema, " +
-          "no idea which columns were dropped, fail to read.")
+          "no idea which columns were dropped, fail to read. Try to disable " +
+          s"${NESTED_SCHEMA_PRUNING_ENABLED.key} to workaround this issue.")
```

Comment:
Although `spark.sql.optimizer.nestedSchemaPruning.enabled` is true by default, can we hide this conf guide when `spark.sql.optimizer.nestedSchemaPruning.enabled=false`, @wangyum ?

Reply:
+1